lib/java/src/org/apache/thrift/TBaseAsyncProcessor.java

```diff
@@ -43,26 +43,19 @@ public TBaseAsyncProcessor(I iface, Map<String, AsyncProcessFunction<I, ? extend
         return Collections.unmodifiableMap(processMap);
     }
 
-    public boolean process(final AsyncFrameBuffer fb) throws TException {
+    @Override
+    public void process(final AsyncFrameBuffer fb) throws TException {
 
         final TProtocol in = fb.getInputProtocol();
         final TProtocol out = fb.getOutputProtocol();
 
-        //Find processing function
+        // Find processing function
         final TMessage msg = in.readMessageBegin();
         AsyncProcessFunction fn = processMap.get(msg.name);
         if (fn == null) {
-            TProtocolUtil.skip(in, TType.STRUCT);
```

Comment:
This should not be removed; if the method cannot be located we need to eat the inbound struct.